eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/RouteBlinding.scala

```diff
@@ -66,26 +121,64 @@ object RouteBlindingEncryptedDataCodecs {
   private val outgoingNodeId: Codec[OutgoingNodeId] = (("length" | constant(hex"21")) :: ("node_id" | publicKey)).as[OutgoingNodeId]
   private val pathId: Codec[PathId] = variableSizeBytesLong(varintoverflow, "path_id" | bytes).as[PathId]
   private val nextBlinding: Codec[NextBlinding] = (("length" | constant(hex"21")) :: ("blinding" | publicKey)).as[NextBlinding]
+  private val paymentRelay: Codec[PaymentRelay] = (
+    ("length" | constant(hex"0a")) ::
+    ("fee_base_msat" | millisatoshi32) ::
+      ("fee_proportional_millionths" | uint32) ::
+      ("cltv_expiry_delta" | cltvExpiryDelta)).as[PaymentRelay]
+  private val paymentConstraints: Codec[PaymentConstraints] = variableSizeBytesLong(varintoverflow,
+    ("max_cltv_expiry" | cltvExpiry) ::
+      ("htlc_minimum_msat" | millisatoshi) ::
+      ("allowed_features" | featuresCodec)).as[PaymentConstraints]
+
+
+  private val messageRelayDataTlvCodec = discriminated[RouteBlindingEncryptedDataTlv].by(varint)
+    .typecase(UInt64(1), padding)
+    .typecase(UInt64(4), outgoingNodeId)
+    .typecase(UInt64(8), nextBlinding)
+
+  val messageRelayDataCodec: Codec[MessageRelayData] = TlvCodecs.tlvStream[RouteBlindingEncryptedDataTlv](messageRelayDataTlvCodec).complete.narrow({
+    case tlvs if tlvs.get[OutgoingNodeId].isEmpty => Attempt.failure(MissingRequiredTlv(UInt64(4)))
+    case tlvs => Attempt.successful(MessageRelayData(tlvs))
+  },{
+    case MessageRelayData(tlvs) => tlvs
+  })
 
-  private val encryptedDataTlvCodec = discriminated[RouteBlindingEncryptedDataTlv].by(varint)
+  private val paymentRelayDataTlvCodec = discriminated[RouteBlindingEncryptedDataTlv].by(varint)
     .typecase(UInt64(1), padding)
     .typecase(UInt64(2), outgoingChannelId)
-    .typecase(UInt64(4), outgoingNodeId)
```

Comment:
Actually you cannot remove this line, otherwise blinded trampoline routes don't work!

Reply:
That's correct, blinded trampoline won't work. We need a lot more spec work to do it. For now I'm ignoring the problem.

Reply:
Fair enough, it's indeed not an issue to disallow it for now since we'll need to add some code to implement blinded trampoline routes anyway, we'll add it when we're ready.